index.bs

```diff
@@ -460,23 +470,42 @@ agent MUST run the following steps:
    initialized to the result of [=DatagramDuplexStream/creating=] a {{DatagramDuplexStream}},
    its [=DatagramDuplexStream/create/readable=] set to |transport|.{{[[IncomingDatagrams]]}},
    and its [=DatagramDuplexStream/create/writable=] set to |transport|.{{[[OutgoingDatagrams]]}}.
-1. If the scheme of |parsedURL| is `https`, [=in parallel=],
-   [=initialize WebTransport over HTTP=].
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[Connection]]</dfn> internal slot initialized to null.
+1. Let |promise| be the result of [=initialize WebTransport over HTTP|initializing WebTransport
+   over HTTP=], with |transport|, |parsedURL| and |dedicated|.
+1. [=Upon fulfillment=] of |promise|, run these steps:
+   1. If |transport|'s {{[[WebTransportState]]}} internal slot is not `"connecting"`, then abort
+      these steps.
+   1. Set |transport|'s {{[[WebTransportState]]}} internal slot to `"connected"`.
+   1. Resolve |transport|'s {{[[Ready]]}} internal slot with {{undefined}}.
+1. [=Upon rejection=] of |promise| with |error|, run these steps:
+   1. Set |transport|'s {{[[WebTransportState]]}} internal slot to `"failed"`.
+   1. Reject |transport|'s {{[[Ready]]}} internal slot with |error|.
+   1. Reject |transport|'s {{[[Closed]]}} internal slot with |error|.
 1. Return |transport|.
 
 <div algorithm="initialize WebTransport over HTTP">
-To <dfn>initialize WebTransport over HTTP</dfn> for a given |transport| and
-|parsedURL|, do the following:
-
-1. If {{WebTransportOptions/serverCertificateFingerprints}} is specified, throw
-   a {{NotSupportedError}} exception.
-1. Establish an HTTP/3 WebTransport session to the address identified by
-   |parsedURL| following the procedures in [[!WEB-TRANSPORT-HTTP3]].
-1. If the session establishment fails, set |transport|'s {{[[WebTransportState]]}}
-   internal slot to `"failed"`, reject |transport|'s {{[[Ready]]}} with a
-   {{TypeError}} and abort these steps.
-1. Set |transport|'s {{[[WebTransportState]]}} internal slot to `"connected"`.
-1. Resolve |transport|'s {{[[Ready]]}} internal slot with {{undefined}}.
+To <dfn>initialize WebTransport over HTTP</dfn>, given a {{WebTransport}} object
+<var>transport</var>, a [=URL record=] |url|, a and boolean |dedicated|, run these steps.
+
+1. Let |promise| be a new promise.
+1. Return |promise| and run the following steps [=in parallel=].
+1. Let |networkPartitionKey| be the result of [=determining the network partition key=] with
+   |transport|'s [=relevant settings object=].
```

Comment:
This should probably happen before you go in parallel, though I doubt anyone will end up confused.

Reply:
Done.